astropy/modeling/mappings.py

```diff
@@ -34,8 +38,8 @@ def mapping(self):
         return self._mapping
 
     def evaluate(self, *args):
-        if len(args) < self.n_inputs:
-            raise TypeError('{0} expects at most {1} inputs; got {2}'.format(
+        if len(args) != self.n_inputs:
+            raise TypeError('{0} expects {1} inputs; got {2}'.format(
```

Comment:
I was thinking this would only be strict (for `Mapping`) if `n_inputs` was specified *explicitly* (otherwise it would fall back on the more flexible behavior).  It would only be strict always for `Identity` (since there `n_inputs` is always explicit.

I'm not sure though if that's best.  It's certainly more flexible.  What do you think?